Scripts/Runtime/Jobs/Collections/DynamicBufferSharedWriteHandle.cs

```diff
@@ -0,0 +1,294 @@
+using Anvil.CSharp.Core;
+using Anvil.Unity.DOTS.Util;
+using System.Collections.Generic;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// A utility class that handles managing access for shared writing (multiple job types writing at the same time)
+    /// so that jobs can be scheduled easily.
+    ///
+    /// ****IMPORTANT NOTE****
+    /// Using this class is very helpful for the specific task of writing to a <see cref="DynamicBuffer{T}"/> in
+    /// parallel from multiple job types and if you're doing that you should already know what you are doing.
+    /// However, there are ways that you won't get the results you expect. They have been listed to aid in debugging
+    /// if needed.
+    ///
+    /// ALL OF THESE SHOULD BE VERY RARE
+    ///
+    /// - If you are adding/removing/reordering Systems and SystemGroups and messing with the PlayerLoop on a frequent
+    /// basis. Ex: Per frame. The <see cref="WorldCache"/> may not update properly and you will have to update manually
+    /// via <see cref="WorldCache.ManualRebuild"/>
+    ///
+    /// - If you are using <see cref="EntityQuery"/>s that your Systems aren't aware of.
+    /// Ex. <see cref="EntityManager.CreateEntityQuery"/> then we may not detect jobs that touch your
+    /// <see cref="IBufferElementData"/>. Always use the <see cref="SystemBase.GetEntityQuery"/> function.
+    ///
+    /// - If you are scheduling jobs outside of a System that operate on <see cref="IBufferElementData"/> we will not
+    /// be aware of it.
+    ///
+    /// - If you are scheduling more than one job in a System that operates on the <see cref="IBufferElementData"/> and
+    /// one uses the Shared Write handle and the other(s) try and do an exclusive write or read, we treat that
+    /// System as only using the Shared Write handle.
+    ///
+    /// - If you are dynamically turning on and off queries (create and/or destroy or the system runs for a while until
+    /// it passes an if check and only THEN you call the <see cref="SystemBase.GetEntityQuery"/> function that operates
+    /// on your <see cref="IBufferElementData"/>, then we will likely not know about it until the
+    /// <see cref="WorldCache"/> is rebuilt. You should do this manually. Better yet, create all possible queries in
+    /// your <see cref="SystemBase.OnCreate"/> so we're aware even if you don't use them until later.
+    ///
+    /// - You may get a false positive if you have a System that has two or more queries in it. One that operates on 
+    /// your <see cref="IBufferElementData"/> (QueryA) and one that doesn't (QueryB). If the logic has the QueryB
+    /// execute but QueryA doesn't, we still see the system as having executed and we'll count it as a point to move
+    /// the handle up when in reality it could have been ignored. 
```

Comment:
That doesn't quite work through because QueryB's components still need to be managed by Unity's internal system. 

So both QueryA (your dynamic buffer data) and QueryB (other components) need to be created with `SystemBase.GetEntityQuery`

But if you have logic in your `OnUpdate` like this:

```
if (someCondition)
{
    RunJob(QueryA);
}
else
{
    RunJob(QueryB);
}
```

The `System` executed. But `QueryA` may not have. If we knew that `QueryA` didn't run, then we could skip this system and possibly get an earlier start time.

Reply:
Riiight, ok. We'll see how this plays out in practice. I guess the only issue is that you're maybe unable to jump in on a shared write. The skipped query's job handle itself won't take time to complete which is nice.